module/VuFind/src/VuFind/OAI/Server.php

```diff
@@ -444,17 +445,9 @@ protected function attachNonDeleted(
         $headerOnly = false,
         $set = ''
     ) {
-        // Get the XML (and display an error if it is unsupported):
-        if ($format === false) {
-            $xml = '';      // no metadata if in header-only mode!
-        } elseif ('oai_vufind_json' === $format && $this->supportsVuFindMetadata()) {
-            $xml = $this->getVuFindMetadata($record);   // special case
-        } else {
-            $xml = $record
-                ->getXML($format, $this->baseHostURL, $this->recordLinkerHelper);
-            if ($xml === false) {
-                return false;
-            }
+        // no metadata if in header-only mode!
+        if ($format !== false) {
```

Comment:
A more concise and exact refactor might be to replace this if statement with:

```
$xml = ($format === false) ? '' : $this->getRecordAsXML($record, $format);
```

It shouldn't make any functional difference since `empty` behaves the same either way, but I feel a little better guaranteeing that `$xml` is defined rather than leaving it unset in some code paths.